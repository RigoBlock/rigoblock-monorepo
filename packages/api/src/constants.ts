export enum NETWORKS {
  MAINNET = '1',
  ROPSTEN = '3',
  KOVAN = '42',
  LOCALHOST = '5777'
}

export const RPC_URLS = {
  [NETWORKS.MAINNET]: 'wss://mainnet.infura.io/ws/',
  [NETWORKS.ROPSTEN]: 'wss://ropsten.infura.io/ws/',
  [NETWORKS.KOVAN]: 'wss://kovan.infura.io/ws/',
  [NETWORKS.LOCALHOST]: 'ws://localhost:8545'
}
