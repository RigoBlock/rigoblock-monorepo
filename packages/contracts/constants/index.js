const path = require('path')

const NETWORKS = {
  ganache: 'http://localhost:8545',
  mainnet: 'https://mainnet.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  ropsten: 'https://ropsten.infura.io/v3/d48872aa1c00471c825e9d856c3c3138',
  kovan: 'https://kovan.infura.io/v3/d48872aa1c00471c825e9d856c3c3138'
}

module.exports = {
  TMP_DIR: '.tmp',
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../src'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'AEthfinex',
    'ATotlePrimary',
    'AWeth',
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Exchange',
    'ExchangeEfx',
    'ExchangeV1Fork',
    'ExchangesAuthority',
    'Faucet',
    'HGetDragoData',
    'Inflation',
    'Migrations',
    'NavVerifier',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'SigVerifier',
    'TokenTransferProxy',
    'TotlePrimary',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WETH9',
    'WrapperLock',
    'WrapperLockEth',
    'ZeroExExchangeHandler'
  ]
}
