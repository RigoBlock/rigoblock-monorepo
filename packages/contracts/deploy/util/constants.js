export const constants = {
    optimizerEnabled: false,
    timeoutMs: 30000,
    contracts: '*' as '*'
}
