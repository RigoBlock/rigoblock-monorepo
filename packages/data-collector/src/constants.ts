export const REDIS_HOST = process.env.REDIS_HOST || '127.0.0.1'
export const REDIS_PORT = parseInt(process.env.REDIS_PORT) || 6379
export const REDIS_PASSWORD = process.env.REDIS_PASSWORD || ''
export const TELEGRAF_HOST = process.env.TELEGRAF_HOST || '127.0.0.1'
export const TELEGRAF_PORT = process.env.TELEGRAF_PORT || 8125

export const TOKEN_MARKET_BASE_URL =
  'https://tokenmarket.net/blockchain/ethereum/assets'
export const CRYPTO_NEWS_BASE_URL =
  'https://cryptopanic.com/api/posts/?auth_token=5b3f861c261e7e6243a55dc7b7ca600832156886&public=true&filter=hot&currencies='
