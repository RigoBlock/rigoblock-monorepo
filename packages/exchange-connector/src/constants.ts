export const MAINNET: string = 'mainnet'
export const ROPSTEN: string = 'ropsten'
export const KOVAN: string = 'kovan'
export const RINKEBY: string = 'rinkeby'

export const PRICE_PRECISION: number = 7
export const AMOUNT_PRECISION: number = 5
export const TO_WEI: number = 1e18

export enum supportedExchanges {
  ETHFINEX = 'Ethfinex',
  ETHFINEX_RAW = 'EthfinexRaw',
  ERCDEX = 'ERCdEX',
  ZEROEXRELAYER = '0xStandardRelayer'
}

export enum NETWORKS {
  MAINNET = 1,
  KOVAN = 42,
  ROPSTEN = 3,
  RINKEBY = 4
}

export enum WS_STATUS {
  CLOSED = 'closed',
  OPEN = 'open',
  CONNECTING = 'connecting',
  CLOSING = 'closing'
}
