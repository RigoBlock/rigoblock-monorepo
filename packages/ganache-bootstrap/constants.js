module.exports = {
  GANACHE_URL: 'http://localhost:8545'
}
