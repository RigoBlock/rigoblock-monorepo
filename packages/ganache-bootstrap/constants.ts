const task = {
  GANACHE_URL: 'http://localhost:8545'
}

export default task
