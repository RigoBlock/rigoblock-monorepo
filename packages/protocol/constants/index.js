const path = require('path')

const NETWORKS = process.env.NETWORKS
  ? process.env.NETWORKS.split(',')
  : ['http://localhost:8545']

module.exports = {
  ARTIFACTS_DIR: path.resolve(__dirname, '../artifacts'),
  CONTRACTS_DIR: path.resolve(__dirname, '../solidity-contracts'),
  NETWORKS,
  GANACHE_PORT: 8545,
  GANACHE_NETWORK_ID: 5777,
  GAS_ESTIMATE: 5e6,
  CONTRACT_NAMES: [
    'Authority',
    'Distribution',
    'Drago',
    'DragoEventful',
    'DragoFactory',
    'DragoRegistry',
    'ERC20',
    'Inflation',
    'Migrations',
    'Owned',
    'OwnedUninitialized',
    'ProofOfPerformance',
    'RigoToken',
    'TokenTransferProxy',
    'UnlimitedAllowanceToken',
    'Vault',
    'VaultEventful',
    'VaultFactory',
    'WrapperLock',
    'WrapperLockEth'
  ]
}
