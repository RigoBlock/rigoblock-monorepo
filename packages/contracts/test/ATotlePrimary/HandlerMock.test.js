import { GANACHE_NETWORK_ID, GAS_ESTIMATE } from '../../constants'
import totlePrimaryArtifact from '../../artifacts/TotlePrimary.json'
import web3 from '../web3'

const contractName = 'TotlePrimary'

describeContract(contractName, () => {
  let totlePrimaryAddress
  let totlePrimaryInstance
  let transactionDefault

  beforeAll(async () => {
    totlePrimaryAddress = await baseContracts[
      'TotlePrimary'
    ].address
    totlePrimaryInstance = new web3.eth.Contract(
      totlePrimaryArtifact.networks[GANACHE_NETWORK_ID].abi,
      totlePrimaryAddress
    )
    transactionDefault = {
      from: accounts[0],
      gas: GAS_ESTIMATE,
      gasPrice: 1
    }
  })

  describe('performRebalance', () => {
    it('performs a ETH-GRG transaction', async () => {
      /*await baseContracts['TotlePrimary'].addHandlerToWhitelist(
        baseContracts['HandlerMock'].address
      )*/ // already whitelisted
      await baseContracts['RigoToken'].transfer(
        baseContracts['HandlerMock'].address,
        48333317481
      )
      await web3.eth.sendTransaction({
        from: accounts[0],
        to: baseContracts['HandlerMock'].address,
        value: 50000
      })

      const isSell = true // false to Eth to Dai, true for Dai to Eth

      //Returns an encoded function call to the primary that has a buy order for the handlerMock
      /*
      function generateEthToDaiRebalance(handlerMock, totlePrimary){
        return generateRebalance(handlerMock, totlePrimary, false)
      }

      function generateDaiToEthRebalance(handlerMock, totlePrimary){
        return generateRebalance(handlerMock, totlePrimary, true)
      }*/

      const encodedOrder = await web3.eth.abi.encodeParameters(
        ['uint256'],
        [2000]
      )

      const txHash = await totlePrimaryInstance.methods.performRebalance(
        [
          [
            isSell,
            baseContracts['RigoToken'].address,
            10000,
            false,
            1,
            10000,
            [
                [baseContracts['HandlerMock'].address, encodedOrder]
            ]
          ]
        ],
        '0x0000000000000000000000000000000000000000',
        '0x1111111111111111111111111111111111111111111111111111111111111111'
      ).send({ ...transactionDefault })
    })
  })
})
