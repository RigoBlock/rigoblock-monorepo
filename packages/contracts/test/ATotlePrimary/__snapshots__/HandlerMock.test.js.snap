// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` TotlePrimary performRebalance performs a ETH-GRG buy transaction 1`] = `"Returned error: VM Exception while processing transaction: revert"`;

exports[` TotlePrimary performRebalance performs a GRG-ETH sell transaction 1`] = `"Returned error: VM Exception while processing transaction: revert"`;
